api/tasks.go

```diff
@@ -430,14 +433,12 @@ func (g *TaskGroup) Canonicalize(job *Job) {
 			MaxDelay:      helper.TimeToPtr(structs.DefaultBatchJobReschedulePolicy.MaxDelay),
 			Unlimited:     helper.BoolToPtr(structs.DefaultBatchJobReschedulePolicy.Unlimited),
 		}
-	default:
-		defaultReschedulePolicy = nil
 	}
 
-	if defaultReschedulePolicy != nil && g.ReschedulePolicy != nil {
+	if defaultReschedulePolicy != nil {
```

Comment:
```
if g.ReschedulePolicy == nil && defaultReschedulePolicy != nil {
  g.ReschedulePolicy = defaultReschedulePolicy
} else if g.ReschedulePolicy != nil {
  g.ReschedulePolicy.Merge(defaultReschedulePolicy)
}
```

And then you can assert that if the policy isn't nil, all fields will be populated and you can put the ApiToStructsJob back to the simple dereferencing